portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -133,7 +149,53 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty("text", request.getParameter("comment"));
     commentEntity.setProperty("date", DateFor.format(new Date()));
     commentEntity.setProperty("timestamp", System.currentTimeMillis());
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, response, "image");
+    if (imageUrl != null) {
+      commentEntity.setProperty("imageURL", imageUrl);
+    }
     datastore.put(commentEntity);
-    response.sendRedirect("index.html");
+    response.sendRedirect("/index.html");
+  }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, HttpServletResponse response, String formInputElementName) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
```

Comment:
Are you sure the user submitted at least one? Sure the form has a single file input, but what if someone tampered with that.

Remember to never trust the client.

Reply:
Well I checked before if the blobKeys array is empty.